src/helpers.js

```diff
@@ -204,3 +204,129 @@ export function getTraversableCells(cpt, map, me) {
   }
   return emptyCells;
 }
+
+
+/*
+ * Initializes and returns a 2D array with the specified width, height, and
+ * default value.
+ *
+ * width: the width of the initialized 2D array
+ * height: the height of the initialized 2D array
+ * defaultVal: the value to give each element in the initialized 2D array
+ */
+export function init2dArray(width, height, defaultVal = 0) {
+  const matrix = [];
+
+  for (let x = 0; x < width; x++) {
+    matrix[x] = new Array(height);
+    for (let y = 0; y < height; y++) {
+      matrix[x][y] = defaultVal;
+    }
+  }
+
+  return matrix;
+}
+
+
+/*
+ * Returns the sum of all corresponding elements from two matrices being
+ * multiplied together. For example:
+ *   a = 1 2   b = 4 1
+ *       3 4       2 3
+ *   multiplyCorrespondingElementsAndSum(a, b)
+ *   (1*4) + (2*1) + (3*2) + (4*3) = 21
+ *
+ * m1: the first matrix
+ * m2: the second matrix (of the same dimensions as m1)
+ */
+export function multiplyCorrespondingElementsAndSum(m1, m2) {
+  const mWidth = m1.length;
+  const mHeight = m1[0].length;
+  let sum = 0;
+
+  for (let x = 0; x < mWidth; x++) {
+    for (let y = 0; y < mHeight; y++) {
+      sum += m1[x][y] * m2[x][y];
+    }
+  }
+
+  return sum;
+}
+
+
+/* Returns a matrix with a buffer applied around its perimeter. Example:
+ *   Take the element values in the corners of the matrix and apply them
+ *   as a bufSize by bufSize matrix to the new corners
+ *   Example:
+ *     m = 1 2 3   bufSize = 2   mWithBuf = 1 1       3 3
+ *         4 5 6                            1 1       3 3
+ *         7 8 9                                1 2 3
+ *                                              4 5 6
+ *                                              7 8 9
+ *                                          7 7       9 9
+ *                                          7 7       9 9
+ *   Take the element values on the edges of the matrix and apply them as
+ *   rows/columns that are bufSize long to the new edges
+ *     mWithBuf = 1 1       3 3   bufSize = 2  mWithBuf = 1 1 1 2 3 3 3
+ *                1 1       3 3                           1 1 1 2 3 3 3
+ *                    1 2 3                               1 1 1 2 3 3 3
+ *                    4 5 6                               4 4 4 5 6 6 6
+ *                    7 8 9                               7 7 7 8 9 9 9
+ *                7 7       9 9                           7 7 7 8 9 9 9
+ *                7 7       9 9                           7 7 7 8 9 9 9
+ *
+ * m: a 2D array, which will have a buffer added around its perimeter
+ * bufSize: the size of buffer to add around the matrix
+ */
+export function addBufferTo2dArray(m, bufSize) {
+  const mWidth = m.length;
+  const mHeight = m[0].length;
+  const mWithBufWidth = mWidth + (bufSize * 2);
+  const mWithBufHeight = mHeight + (bufSize * 2);
+
+  // Create a grid with matrix's values and a bufSize perimeter of zeros
+  const mWithBuf = init2dArray(mWithBufWidth, mWithBufHeight);
```

Comment:
you should pass 0 as an input to the function (what if its default behavior changes in the future?)